pkg/querier/blocks_store_queryable.go

```diff
@@ -257,11 +250,17 @@ func NewBlocksStoreQueryableFromConfig(querierCfg Config, gatewayCfg storegatewa
 		return nil, errors.Wrap(err, "failed to create store set")
 	}
 
+	maxReplication := gatewayCfg.ShardingRing.ReplicationFactor
+	if gatewayCfg.DynamicReplication.Enabled {
+		maxReplication *= gatewayCfg.DynamicReplication.Multiple
```

Comment:
To keep the logic together, maybe add a `MaxReplication()` method to the dynamic replication interface/object above. It'd also be nice if the no-op one returned the default replication factor so we don't use extra retries when they're not needed.

Reply:
have a look at https://github.com/grafana/mimir/pull/11354/commits/8293fb6119a79372df7682e90fe4cc66064e21ff

i wasn't sure what's the best way to wire it up. Another option would be to expose this on the `BlocksStoreSet` interface, but i think that's a bit out of its concern. PTAL 🙏